[CS2113-T13-1] MathGenius

yzhanglp opened the pull request:
MathGenius is an automated arithmetic and equation problem generator that offers a high degree of customization. It will calculate the accuracy and time needed as a record to track your performance.

classskipper351 committed:
增加了生成题目的功能， 接受数量，运算符号和最大的位数三个参数，（有默认参数），并生成题目，保存在test中。
题目暂时有description和answer的部分，出题的时候不把answer打印出来
We added a feature to generate questions. It accepts three parameters: the number of questions, arithmetic operators, and maximum number of digits (with default parameters). It generates questions and saves them in 'test'. Each question currently consists of a description and an answer. When generating questions, the answers are not printed.

classskipper351 committed:
Merge pull request #6 from classskipper351/generate_problems

生成问题generate problems

yzhanglp committed:
add the checker and checkTest

Geinzit committed:
Fix pathing in AboutUs.md. Add class Storage and Record for data persistance feature.

Andy123qq4 committed:
Default UI version: get input command from ProblemGenerator; Add UI.java and Parser.java

yzhanglp committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Andy123qq4 committed:
Little bug fix on ProblemGen and UI

Geinzit committed:
Refine code style. Add read file in main.

Geinzit committed:
Merge branch 'master' into master

Geinzit committed:
Merge pull request #7 from Geinzit/master

Fix pathing in AboutUs.md. Add class Storage and Record for data pers…

classskipper351 committed:
add test class

classskipper351 committed:
Merge branch 'master' into generate_problems

classskipper351 committed:
Merge pull request #13 from classskipper351/generate_problems

add test class

classskipper351 committed:
solve merge conflicts

classskipper351 committed:
Merge pull request #14 from classskipper351/generate_problems

solve merge conflicts

yzhanglp committed:
fix some bugs in checker

yzhanglp committed:
merge upstream

Andy123qq4 committed:
Fix code style and typos.

Andy123qq4 committed:
Merge pull request #15 from Andy123qq4/master

Fix code style and typos.

yzhanglp committed:
update checker to receive user input answer

yzhanglp committed:
merge upstream and fix conflict

Andy123qq4 committed:
Code style formatting & UI help function fix

Andy123qq4 committed:
Code style formatting & UI help function fix

classskipper351 committed:
add message when using default parameters and add my profile
使用默认参数时添加了消息并添加了我的配置文件

yzhanglp committed:
Add the time

classskipper351 committed:
Merge pull request #17 from classskipper351/master

add message when using default parameters and add my profile

yzhanglp committed:
Merge pull request #18 from yzhanglp/master

Add the checker class

Andy123qq4 committed:
Merge branch 'master' of https://github.com/Andy123qq4/tp

classskipper351 committed:
code style problems

Andy123qq4 committed:
Gradle + UI

classskipper351 committed:
Merge pull request #19 from classskipper351/master

code style problems

Andy123qq4 committed:
Merge branch 'master' into master

Andy123qq4 committed:
Merge pull request #20 from Andy123qq4/master

Gradle + UI

Geinzit committed:
Fix parsing bug in Storage. Integrate Storage write-to-file function in parser.

Geinzit committed:
Merge branch 'master' into branch-heinzhuang

Geinzit committed:
fix code style to fit standard.

Geinzit committed:
resolve conflicts.

Geinzit committed:
merge files.

Geinzit committed:
Merge pull request #21 from Geinzit/branch-heinzhuang

Fix parsing bug in Storage. Integrate Storage write-to-file function …

Geinzit committed:
fix bug so that parameters will properly load.

Geinzit committed:
Merge pull request #22 from Geinzit/branch-heinzhuang

fix bug so that parameters will properly load.

Celineyaa committed:
Add a Jar file

Celineyaa committed:
Add JUnit tests for ProblemGenerator, to test whether the generated problem sets' operator types, numbers of problem and digits are match with the user input.

Celineyaa committed:
Adapted according to checkstyle

Celineyaa committed:
Adapt the checkstyle again

Celineyaa committed:
Merge pull request #23 from Celineyaa/master

Add JUnit test for ProblemGenerator

yzhanglp committed:
update the DG with userstudy and etc.

yzhanglp committed:
Merge pull request #25 from yzhanglp/master

update the DG with userstudy and etc.

Geinzit committed:
add Record and Storage entries to the DG.

Geinzit committed:
Merge pull request #26 from Geinzit/branch-heinzhuang

add Record and Storage entries to the DG.

yzhanglp committed:
add the implematation of checker

yzhanglp committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

yzhanglp committed:
Merge pull request #27 from yzhanglp/master

Add the checker component

Celineyaa committed:
Add test case and Instructions for manual testing in DG

Celineyaa committed:
Merge branch 'master' into master

Celineyaa committed:
Merge pull request #28 from Celineyaa/master

Add test case and Instructions for manual testing in DG

classskipper351 committed:
update UML diagram and DG

classskipper351 committed:
Merge pull request #29 from classskipper351/master

update UML diagram and DG

Geinzit committed:
Add Record viewing feature, with multiple viewing and sorting options. Minor adjustments and fixes.

classskipper351 committed:
增加了长算式功能， Test类新增操作数参数 ，新增了一个计算器类来计算算式
The long calculation function has been added, the operand parameter has been added to the Test class, and a calculator class has been added to calculate the formula

classskipper351 committed:
Merge pull request #35 from classskipper351/long_formula

增加了长算式功能， Test类新增操作数参数 ，新增了一个计算器类来计算算式

Celineyaa committed:
Update the testcase part of DG

Geinzit committed:
Merge pull request #34 from Geinzit/branch-heinzhuang

Add Record viewing feature, with multiple viewing and sorting options…

Celineyaa committed:
Merge pull request #1 from AY2324S2-CS2113-T13-1/master

Update the part of the testcase in DG

Celineyaa committed:
Update the User guide of version 1

Celineyaa committed:
no message

Celineyaa committed:
Update the UserGuide of ver1

Celineyaa committed:
Remove UserGuide.md

Celineyaa committed:
Update User Guide for version1

Celineyaa committed:
Merge pull request #36 from Celineyaa/master

Update the version 1 user guide

yzhanglp committed:
Let the checker to show the incorrect answer

yzhanglp committed:
Merge branch 'master' into master

yzhanglp committed:
Merge pull request #37 from yzhanglp/master

Checker V2.0

Geinzit committed:
update Developer Guide regarding Record and Storage. Remove magic numbers. Minor bug fixes.

Geinzit committed:
Merge pull request #39 from Geinzit/branch-heinzhuang

update Developer Guide regarding Record and Storage. Remove magic num…

Geinzit committed:
update UserGuide with better formatting and additional information.

Geinzit committed:
Merge pull request #41 from Geinzit/branch-heinzhuang

update UserGuide with better formatting and additional information.

Celineyaa committed:
Correct the gradle errors

Celineyaa committed:
Merge pull request #42 from Celineyaa/master

Correct the gradle errors

Geinzit committed:
improve code style.

Geinzit committed:
Merge pull request #43 from Geinzit/master

improve code style.

classskipper351 committed:
修改了我的部分的DG， 修改了UML类图
Modified my part of the DG, modified the UML class diagram

classskipper351 committed:
Merge branch 'master' into master

classskipper351 committed:
Merge pull request #44 from classskipper351/master

修改了我的部分的DG， 修改了UML类图

yzhanglp committed:
add the sequence diagram, fix some bugs in UG

yzhanglp committed:
Merge pull request #45 from yzhanglp/master

add the sequence diagram, fix some bugs in UG

Celineyaa committed:
no message

Celineyaa committed:
Increase and updated testcases for ProblemGenerator, Checker, UI, Storage.
Add an accessor of records in Storage.
Modify a typo in Storage.

Andy123qq4 committed:
UI intergration v2 + formatting

Andy123qq4 committed:
UI intergration v2 + formatting

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/seedu/duke/Record.java

Andy123qq4 committed:
Merge pull request #47 from Andy123qq4/master

UI intergration v2 + formatting

Andy123qq4 committed:
UI intergration v2 + formatting

Andy123qq4 committed:
UI intergration v2 + formatting

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

Andy123qq4 committed:
Merge pull request #48 from Andy123qq4/master

Fix checker and bug

Andy123qq4 committed:
Fix missing photo on DG: Move asset/image to DG folder

Andy123qq4 committed:
Fix missing photo on DG: Move asset/image to DG folder

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

Celineyaa committed:
Remove trial.java

tayponghee commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+### The overall Sequence Diagram:
+![Sequence Diagram](../assets/images/sequenceDiagram.png)    
+
```
Please remove these extraneous empty liens

Andy123qq4 committed:
Merge pull request #49 from Andy123qq4/master

Fix missing photo on DG: Move asset/image to DG folder

Celineyaa committed:
Adapted the code suggested by gradle.

Celineyaa committed:
Merge pull request #46 from Celineyaa/master

Update testcases, Add an accessor in Storage

mukund1403 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
+
```
Can consider omitting void in methods under Class diagrams to improve readability. 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/033dda2f-e28d-4d63-a650-34578c45efce)

You can consider referring to example given here for the method setNumber:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/69431793-7560-40b0-b6cc-aff80de0ee05)

mukund1403 replied:
Could the symbols you used to show visibility of attributes and methods be wrong perhaps? 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/033dda2f-e28d-4d63-a650-34578c45efce)

Refer to the example given here:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/bd557ef1-06b1-4400-855b-0a9dffe14664)

mukund1403 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+
+
+### problemGenerator Component
+API: [ProblemGenerator.java](../src/main/java/seedu/duke/ProblemGenerator.java)
+
```
Are these links functional? Perhaps you could provide functional links to the code or omit them if not necessary.
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/108778858/83eb12ef-88f8-4253-a451-d6a29eedcdd6)

mukund1403 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+1. Support check on more types of problems (i.e Quadratic equation of one variable)   
+2. Give some explanation of the math problems.   
+3. Store the answer in array format to hand the function and matrix answer.
+   
+* To be added
+
```
Could this be a redundant line? Maybe you can consider deleting it.

Cohii2 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
```
Is the (c) icon in UML diagram meant to be there? Please consider removing it. 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111263799/ed5f8488-fae4-4403-86b7-25f499f0ddbf)

Cohii2 replied:
Arrows look a little messy - maybe neaten them up a bit?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111263799/c64b0d2a-fb84-45fd-8c0a-1587b2a603bb)

yeozhishen replied:
it would be good if multiplicities are specified

LWachtel1 replied:
Should you have everything in this UML diagram? Perhaps could reduce number of methods and members listed to make a summary UML diagram then put more detailed smaller diagrams in the component sections

Cohii2 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+### Record Component - Design
+
+API: [Record.java](../src/main/java/seedu/duke/Record.java)
```
API link not working - maybe look into this?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111263799/ef6f87c9-c21f-4553-b6dd-5aec31f54137)

Cohii2 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+5. The checker will also store the times that user use to caculate for the problemset.  
+6. The checker will also store the questions that user gave a wrong answer as well as the wrong answer.
+    
+**`Pseudo code` for reference:**  
```
Good inclusion of pseudo code to make DG more understantable!

HenryGan138 replied:
Really like your idea of including pseudo-code, making it much clear and easy to understand.

Cohii2 reviewed (commented):
Overall some bugs but I like the level of detail.

pqienso commented on docs/assets/images/sequenceDiagram.png:
![Screenshot 2024-04-05 130821](https://github.com/nus-cs2113-AY2324S2/tp/assets/85116669/90fad4ff-2ba3-41e8-9c7b-9e24317d8d35)
Should these self-invoked methods have an overlapping activation bar, with a dotted return line?
Here's an example:
![Screenshot 2024-04-05 131000](https://github.com/nus-cs2113-AY2324S2/tp/assets/85116669/9f94ba9e-e476-4db8-8f15-97e6912644fe)

pqienso commented on docs/assets/images/sequenceDiagram.png:
![Screenshot 2024-04-05 131152](https://github.com/nus-cs2113-AY2324S2/tp/assets/85116669/feedadef-63e9-41d4-bebe-94bcd08fcae5)
Consider adding the dotted line to represent the program returning. For example:
![Screenshot 2024-04-05 131255](https://github.com/nus-cs2113-AY2324S2/tp/assets/85116669/48090477-77a7-434f-a4d8-f580cd689b62)

pqienso commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+- Store in each Record object an ArrayList of Problems objects for storing specifics of the attempted Problem Set.
+- When a problemSet solving is saved to a record **for the first time**, the corresponding Record object will create a unique ID for the problem set using Java's built-in hashCode method. When loading this record in the future and re-saving the data, the same ID will be used and no new IDs will be generated. This is achieved by using two different constructors for these two different situations.
+
+**Code Snippet**
```
Consider replacing the code snippets with a sequence diagram of the record component to help with visualization

pqienso commented on docs/assets/images/sequenceDiagram.png:
Can this sequence diagram be broken down into several smaller sequence diagrams spread throughout the DG? This could help with easier visualization and reading.

pqienso commented on docs/assets/images/mathGenius.png:
Consider removing some of the less important methods to reduce clutter

yeozhishen commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+
+### The overall Sequence Diagram:
+![Sequence Diagram](assets/images/sequenceDiagram.png)    
```
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110598041/b277e220-5921-453a-8e42-bafbdfab2872)
Is the activation bar is supposed to end mid sequence for duke? or is it supposed to continue all the way down

yeozhishen replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110598041/4d920459-5c65-407a-92e9-23c05c055f69)
does not return function control back to the caller, same goes for a few other function calls in the same sequence diagram

mihirheda02 replied:
Shouldn't your `readCommand()` function be inside the main loop?

LWachtel1 replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/68402849/4c68dca3-fdda-4ad1-ad39-01f6e88e6a4b)
Could an action bar be missing for writeFile()?

LWachtel1 replied:
Could you perhaps break large sequence diagram into smaller, simpler sequence diagrams that can be placed in each component section?

pqienso reviewed (commented):
DG looks good overall, but consider breaking down the UML diagrams into smaller diagrams and spread them out throughout the DG to make it more readable

yeozhishen commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+when the gengerator is triggered, users enter a command like this:
+
+    generate -t [operators] -n [number] -d [maximum digit] -l[length]
+for example , user can enter:
```
wrong display? the "for example, user can enter: " part is inside the code blocks

mihirheda02 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+### Storage Component - Design
+
+API: [Storage.java](../src/main/java/seedu/duke/Storage.java)
```
I don't think the link to the code is working here

yeozhishen commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
 ### Target user profile
 
-{Describe the target user profile}
+* The student who wish to practise their mathematical problem solving ability.  
```
change to students to be more in line with the next statement?

mihirheda02 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+in which case the parseCommand() will take the command and pass the parameters to generete().
+
+generate() function will choose 3 random operand with max digits of 2, take a random operation from the operation set of  $+,-,*,/$ to form a single problem ,and loop the execution for 10 times to form a test
```
Not sure if the dollar signs in `$+,-,*,/$` were meant for formatting, but it looks a little confusing.

yeozhishen commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+- a snapshot of a completed problem sets, including the individual problems, the date it was solved, the time taken to solve the problem set, and the accuracy of the attempt.
+- records the specifics of each problem, the unique ID(created using Java's default hashcode) of the problem set, for organized or filtered viewing of past records.
+- For the viewing of a record, users may choose whether to display the problem set specifics or not using the `showProbDetails` parameter.
+- When writing a record to external file, it's written into a single line in the format of "/dateTime(format:yyyy-MM-dd HH:mm:ss) /speed /accuracy /problemSetID /problems". For problems, each problem is separated by a space and is formatted in "/problemDescription,/problemAnswer"
```
it would be good to give an example for reference

mihirheda02 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|v1.0|student who just started learning quadratic equations|use MathGenius to specifically generate problem sets|gain specialized practice on this topic|
+|v1.0|a student who’s unfamiliar with mathematics terms|watch explanations and introductions to unfamiliar terms|strengthen understanding|
+|v1.0|primary school teacher|wcreate course or topic-specific arithmetic questions for students|teaching student more effortlessly|
```
Is `wcreate` a typo?

mihirheda02 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+## Launch and Shutdown
+   1. initial launch:
+      1. Download the jar file and copy into an empty folder
+      2. run the jar file use command '_java -jar tp.main.jar_'
```
It might be more readable to use markdown code block for '_java -jar tp.main.jar_'

LWachtel1 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+    generate -t [operators] -n [number] -d [maximum digit] -l[length]
+for example , user can enter:
+
+    generate -t +-*/ -n 10 -d 2 -l 3
```
Could you perhaps include output for this sample input?

LWachtel1 reviewed (commented):
Large level of detail on UML and sequence diagrams shows good understanding of program, but could perhaps could consider splitting into smaller diagrams for each component section for more manageable visualisation

HenryGan138 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
```
<img width="114" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/86831302/7f0bbbcf-66a9-496c-be6d-9bfb9bacbc6e">
Nice jobs on your uml diagram, but you can replace the red square and green circle with "+" and "-" sign.

HenryGan138 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+![UML Diagram](assets/images/mathGenius.png)    
+
+
+### The overall Sequence Diagram:
```
<img width="662" alt="image" src="https://github.com/nus-cs2113-AY2324S2/tp/assets/86831302/afcf0802-7d7b-4e62-8435-24e38c2a6991">
It seems this SD has included the entire program works. If it will be better you break each part down like storage, Ui, etc..

HenryGan138 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+The Record component:
+
+- a snapshot of a completed problem sets, including the individual problems, the date it was solved, the time taken to solve the problem set, and the accuracy of the attempt.
```
Yes, probably it is better if you can add some snapshots to explain

HenryGan138 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+
+### Storage Component - Implementation
+
+- Uses a list of Record objects to store all past attempts
```
Probably it is better if you can include the code of Record class here

HenryGan138 commented on docs/DeveloperGuide.md:
```diff
@@ -6,33 +6,230 @@
+    ProblemGeneratorTest#parseCommand() — parse the input command to fit in the program.
+    ProblemGeneratorTest#parseNumber() — parse out the operands from a given problem.
+
+Given below is an example usage scenario and how the test behaves.
```
Maybe you can add some snapshots here to help your illustration

HenryGan138 reviewed (commented):
Overall, I love your DG, detailed and clear.

Geinzit committed:
Update UserGuide.md, better highlighting for records display format.

Geinzit committed:
Update help command to show explanation for 'exit' command.

Geinzit committed:
Merge pull request #81 from Geinzit/master

Update help command to show explanation for 'exit' command.

Geinzit committed:
Update help command's default display

Geinzit committed:
Merge branch 'AY2324S2-CS2113-T13-1:master' into master

Geinzit committed:
Merge pull request #82 from Geinzit/master

Update help command's default display

Geinzit committed:
add 'retry' feature to retry priviously attempted problem sets. Remove some redundant code and some ui adjustments.

Geinzit committed:
fix to switch case. Display speed as 'problems per minute'

Geinzit committed:
Merge pull request #83 from Geinzit/master

add 'retry' feature to retry priviously attempted problem sets. Remov…

yzhanglp committed:
add the explanation

Andy123qq4 committed:
invalid input

Andy123qq4 committed:
Merge branch 'master' into master

Andy123qq4 committed:
Merge pull request #84 from Andy123qq4/master

invalid input

Geinzit committed:
update text-ui-test content for Java CI tests

Geinzit committed:
fix gradle reported issues.

Geinzit committed:
removed chinese comments.

Geinzit committed:
removed more chinese comments.

Andy123qq4 committed:
Fix AboutUs.md

Andy123qq4 committed:
Fix wrong spelling

classskipper351 committed:
added validity check of generation input number
增加了输入的合法性检查

classskipper351 committed:
Merge pull request #86 from classskipper351/master

added validity check of generation input number

Geinzit committed:
removed all \n and \t signs in tests.

Geinzit committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Andy123qq4 committed:
Merge pull request #87 from Andy123qq4/master

Fix PE-D UserGuide & DeveloperGuide

Geinzit committed:
fix illegal forwarding bug.

yzhanglp committed:
add the simple explanation

Geinzit committed:
fix code style for gradle

Geinzit committed:
Merge pull request #85 from Geinzit/master

update text-ui-test content for Java CI tests and more fixes

yzhanglp committed:
fix merge conflict

yzhanglp committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Geinzit committed:
remove \r symbols during testing.

Geinzit committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Geinzit committed:
add clean output to all Uitests

Geinzit committed:
Merge pull request #88 from Geinzit/master

remove \r symbols during testing.

Andy123qq4 committed:
Move all UML diagram to another folder

Andy123qq4 committed:
Minor adjustments

Andy123qq4 committed:
Better readability in UI and better help function

Andy123qq4 committed:
Enhance and add more detail for 2 guides

Andy123qq4 committed:
Enhance and add more detail for 2 guides

Andy123qq4 committed:
Merge branch 'master' of https://github.com/Andy123qq4/tp

Andy123qq4 committed:
Fix Ui and UiTest for gradle checking

classskipper351 committed:
drafted my ppp

classskipper351 committed:
Merge pull request #89 from classskipper351/master

drafted my ppp

Geinzit committed:
add PPP for user Geinzit

Geinzit committed:
Merge pull request #90 from Geinzit/master

add PPP for user Geinzit

Andy123qq4 committed:
Merge branch 'master' into master

Geinzit committed:
adjust file paths for portfolios

Andy123qq4 committed:
Merge pull request #91 from Andy123qq4/master

Fix PE-D bugs

yzhanglp committed:
add the ppp and some parts of explanation generator

Geinzit committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

yzhanglp committed:
Merge pull request #92 from yzhanglp/master

Add PPP and first versin of explanation generator

Andy123qq4 committed:
Add first time generate problem instruction

Geinzit committed:
Merge pull request #93 from Geinzit/master

Adjust file paths

Geinzit committed:
fix file paths in AboutUs.md

Geinzit committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Geinzit committed:
Merge pull request #94 from Geinzit/master

fix file paths in AboutUs.md

Andy123qq4 committed:
Merge branch 'AY2324S2-CS2113-T13-1:master' into master

Andy123qq4 committed:
Drafted PPP for Andy123qq4

Andy123qq4 committed:
Merge branch 'master' of https://github.com/Andy123qq4/tp

Andy123qq4 committed:
Fix aboutus when moving files

Andy123qq4 committed:
Delete duplicates

Andy123qq4 committed:
Delete duplicates

Andy123qq4 committed:
Merge branch 'master' of https://github.com/Andy123qq4/tp

Andy123qq4 committed:
Fix Gradle Syntax

Andy123qq4 committed:
Merge pull request #95 from Andy123qq4/master

Drafted PPP for Andy123qq4 & Add first time generate problem instruction & Gradle Fix

Andy123qq4 committed:
Update README.md

Andy123qq4 committed:
Merge pull request #96 from Andy123qq4/master

Update README.md

classskipper351 committed:
修改了calculator的错误，并且现在输入错误格式答案会被强制要求重新输入
Fixed a bug in the calculator and now forces you to re-enter an incorrectly formatted answer

classskipper351 committed:
Merge pull request #97 from classskipper351/master

Fixed a bug in the calculator and now forces you to re-enter an incorrectly formatted answer

Celineyaa committed:
Add class DIYProblemSet and ProblemSetType, modify Storage, Record, Ui and Test to show the ProblemSetType, add PPP draft and modify the UG and DG.

Celineyaa committed:
Merge pull request #98 from Celineyaa/master

Add class DIYProblemSet and ProblemSetType, modify Storage, Record, U…

Celineyaa committed:
Update the code according to gradle tests.

Celineyaa committed:
Merge pull request #99 from Celineyaa/master

Update the code according to gradle tests.

Geinzit committed:
fix spacing errors for text-ui-test

Geinzit committed:
Merge pull request #100 from Geinzit/master

fix spacing errors for text-ui-test as well as variable naming for DIY

yzhanglp committed:
add the explanation and other function, round the answer up to 3 decimal

yzhanglp committed:
Merge pull request #101 from yzhanglp/master

add the explanation and other function, round the answer up to 3 decimal

Celineyaa committed:
no message

Celineyaa committed:
update the DIY user guide

Celineyaa committed:
Merge pull request #102 from Celineyaa/master

Update the DIY user guide

Celineyaa committed:
Fixed the error of infinite output in DIYProblemSet

Celineyaa committed:
Merge pull request #103 from Celineyaa/master

Fixed the error of infinite output in DIYProblemSet

Geinzit committed:
Fix CheckStyle issues. Update UG and DG for 'retry'. Fix Grammar

Geinzit committed:
Merge pull request #104 from Geinzit/master

Fix CheckStyle issues. Update UG and DG for 'retry'. Fix Grammar

Andy123qq4 committed:
Update UserGuide.md with grammarly check

Andy123qq4 committed:
Update DeveloperGuide.md

Celineyaa committed:
Update AboutUs.md and upload image

Celineyaa committed:
Merge pull request #105 from Celineyaa/master

Update AboutUs.md and upload image

Celineyaa committed:
Update celineyaa.md

Celineyaa committed:
Update celineyaa.md

Celineyaa committed:
Merge pull request #106 from Celineyaa/master

Update celineyaa.md

Celineyaa committed:
Update celineyaa.md

Celineyaa committed:
Merge pull request #107 from Celineyaa/master

Update celineyaa.md

classskipper351 committed:
改了DIY的答案检查和部分UI
Changed DIY answer checking and part UI

classskipper351 committed:
Merge pull request #108 from classskipper351/master

改了DIY的答案检查和部分UI

Celineyaa committed:
no message

yzhanglp committed:
Modified the sequence diagram

Andy123qq4 committed:
Clear unused comment, refine syntax and coding standards

Andy123qq4 committed:
Clear unused comment, refine syntax and coding standards

Andy123qq4 committed:
Merge remote-tracking branch 'origin/master'

Andy123qq4 committed:
Avoid * imports

Andy123qq4 committed:
Merge pull request #109 from Andy123qq4/master

Clear unused comment, refine syntax and coding standards

Celineyaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp:
  改了DIY的答案检查和部分UI Changed DIY answer checking and part UI

# Conflicts:
#	recordList.txt

Andy123qq4 committed:
add code segment formatting for functions and class names, data type, variable names. Make sure the capitals and lower case letters are all in a same setting, enhance readbility

Andy123qq4 committed:
Add table of contents

Andy123qq4 committed:
Merge pull request #110 from Andy123qq4/master

DG: Enhance readability by adding code segment formatting for functions, class names, data types, and variables

yzhanglp committed:
add the interface message to UI, give a more detail instruction about checker

yzhanglp committed:
merge the upstream

yzhanglp committed:
add the message before preview the testsets

yzhanglp committed:
Some coding standard bugs changes

Geinzit committed:
delete useless files. update DG and README.md

yzhanglp committed:
fix the gradle problem

Geinzit committed:
merge conflicts

yzhanglp committed:
Merge pull request #112 from yzhanglp/master

Update PPP, DG sequence diagram and some interface message

Celineyaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp:
  Avoid * imports
  Clear unused comment, refine syntax and coding standards
  Clear unused comment, refine syntax and coding standards

Geinzit committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

Geinzit committed:
Merge pull request #113 from Geinzit/master

delete useless files. update DG and README.md

Celineyaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp:
  fix the gradle problem
  delete useless files. update DG and README.md
  Some coding standard bugs changes
  add the message before preview the testsets
  Add table of contents
  add code segment formatting for functions and class names, data type, variable names. Make sure the capitals and lower case letters are all in a same setting, enhance readbility
  Modified the sequence diagram

Celineyaa committed:
solve the infinite loop in DIY class, replace all scanner by ui.readcommand

Celineyaa committed:
Merge pull request #114 from Celineyaa/master

solve the infinite loop, replace scanner by ui.readcommand

classskipper351 committed:
修改了我的代码风格以及exit的bug
Fixed my code style and  bugs in exit

Celineyaa committed:
remove the scanner import

Celineyaa committed:
Merge pull request #115 from Celineyaa/master

remove the scanner import

classskipper351 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	recordList.txt
#	src/main/java/seedu/duke/Calculator.java
#	src/main/java/seedu/duke/Checker.java
#	src/main/java/seedu/duke/ProblemGenerator.java

Andy123qq4 committed:
Fix coding standards: assert, inconsistent, magic number and bool condition format, delete redundant code

classskipper351 committed:
Merge branch 'master' of https://github.com/classskipper351/tp

classskipper351 committed:
Merge pull request #116 from classskipper351/master

Changed the style of my code and fixed bugs

classskipper351 committed:
gradle check passed

Celineyaa committed:
no message

classskipper351 committed:
Merge pull request #117 from classskipper351/master

gradle check passed

Celineyaa committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp

* 'master' of https://github.com/AY2324S2-CS2113-T13-1/tp:
  gradle check passed
  修改了我的代码风格以及exit的bug Fixed my code style and  bugs in exit

# Conflicts:
#	recordList.txt

Andy123qq4 committed:
add javadoc comments and fix minor bugs

yzhanglp committed:
add the exp to retry function

yzhanglp committed:
Merge pull request #119 from yzhanglp/master

add the explanation to retry function

Andy123qq4 committed:
Merge branch 'master' into master

Celineyaa committed:
Update the uml of DIYProblemSet

Andy123qq4 committed:
Merge pull request #118 from Andy123qq4/master

Fix coding standards and add javadoc comments

Celineyaa committed:
Merge pull request #120 from Celineyaa/master

Update the DIYProblemSet UML

Andy123qq4 committed:
Fix ui and ui uml function

Andy123qq4 committed:
Merge pull request #121 from Andy123qq4/master

Fix ui and ui uml function

Andy123qq4 committed:
Rename uml file

Andy123qq4 committed:
fix Get Started:

Andy123qq4 committed:
Merge branch 'master' of https://github.com/Andy123qq4/tp

Andy123qq4 committed:
Merge pull request #122 from Andy123qq4/master

fix Get Started in UG

classskipper351 committed:
add my PPP in my github name
